### PR TITLE
fix: Object spread issue

### LINKED ISSUE #48: SyntaxError: Unexpected token ...
Today i fetch the newest code. and got this error:

```bash
➜ GITHUB_TOKEN=xx node --harmony server.js        
/data1/home/dongweiming/mention-bot2/server.js:120
        try { repoConfig = {...repoConfig, ...JSON.parse(configRes)}; } catch (e) {}
                            ^^^

SyntaxError: Unexpected token ...
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:404:25)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Function.Module.runMain (module.js:457:10)
    at startup (node.js:136:18)
    at node.js:972:3
➜ node --version
v5.0.0
```

I am very confused

### PR DESCRIPTION
**Pull request by dflynn15**
- Updates dependencies with the object spread transform
- Adds plugins to babelrc
- Updates `npm start` to include the babel transform

Fixes #48

I opted with using Babel to transform the `server.js` instead of `babel-node` because as noted in Babel's documentation it is:
> Not meant for production use

Nit-pick away! :smile:

### CODE REVIEW

**Comment by agubler** [comment]
What are your thoughts on abstracting the `require('babel-core/register');` into a top level file and then a simple `require('./server');` would suffice?

Works outputting the compiled `server.js` also, I'm not sure which is the preferred method for here.

**Comment by vjeux** [comment]
Agreed with @agubler suggestion

**Comment by dflynn15** [comment]
I'll refactor :+1:

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules
 npm-debug.log
+server-compiled.js
```
Comment (vjeux) [inline .gitignore]: Remove

File: run-server.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+require('babel-core/register');
+require('./server.js');
```
Comment (vjeux) [inline run-server.js]: Good thanks

**Pull request merged**